extensions/ql-vscode/src/view/compare/Compare.tsx

```diff
@@ -53,6 +61,87 @@ export function Compare(_: Record<string, never>): React.JSX.Element {
           case "setComparisons":
             setComparison(msg);
             break;
+          case "streamingComparisonSetup":
+            setComparison(null);
+            streamingComparisonRef.current = msg;
+            break;
+          case "streamingComparisonAddResults": {
+            const prev = streamingComparisonRef.current;
+            if (prev === null) {
+              console.warn(
+                'Received "streamingComparisonAddResults" before "streamingComparisonSetup"',
+              );
+              break;
+            }
+
+            if (prev.id !== msg.id) {
+              console.warn(
+                'Received "streamingComparisonAddResults" with different id, ignoring',
+              );
+              break;
+            }
+
+            let result: QueryCompareResult;
+            switch (prev.result.kind) {
+              case "raw":
+                if (msg.result.kind !== "raw") {
+                  throw new Error(
+                    "Streaming comparison: expected raw results, got interpreted results",
+                  );
```

Comment:
Any danger of getting here if you switch in the middle of rendering a very large comparison result?

Reply:
There shouldn't be: https://github.com/github/vscode-codeql/pull/3771/commits/510a269dde893d3fd7eb4dafdccecf5532071d97 adds a check for that race condition by adding an `id` to every message so that if a different setup message is received, it will ignore all future messages from the previous stream.